diff --git a/components.ts b/components.ts
index fbf40d1a3b9d2becef6a6282dd16a0e2f9fe26d0..a7145e00b22b846b8fefb22b62f1d584fe84078b 100644
--- a/components.ts
+++ b/components.ts
@@ -1,11 +1,12 @@
-export { Authors, type AuthorsProps } from './app/components/Authors.js'
-export { BlogPosts } from './app/components/BlogPosts.js'
-export { Button } from './app/components/Button.js'
-export { Callout, type CalloutProps } from './app/components/Callout.js'
-export * as HomePage from './app/components/HomePage.js'
-export { Raw } from './app/components/Raw.js'
-export { Sponsors } from './app/components/Sponsors.js'
-export { Steps, type StepsProps } from './app/components/Steps.js'
-export { Step, type StepProps } from './app/components/Step.js'
+export { Authors, type AuthorsProps } from './app/components/Authors.js';
+export { BlogPosts } from './app/components/BlogPosts.js';
+export { Button } from './app/components/Button.js';
+export { Callout, type CalloutProps } from './app/components/Callout.js';
+export * as HomePage from './app/components/HomePage.js';
+export { Link } from './app/components/Link.js';
+export { Raw } from './app/components/Raw.js';
+export { Sponsors } from './app/components/Sponsors.js';
+export { Steps, type StepsProps } from './app/components/Steps.js';
+export { Step, type StepProps } from './app/components/Step.js';
 
-export { components as MDXComponents } from './app/components/mdx/index.js'
+export { components as MDXComponents } from './app/components/mdx/index.js';
diff --git a/package.json b/package.json
index 2708a3e628c6000c52778d6361eceacf22ab9d4d..cbd03443f9b79c39f88c34bb721a87e3de074221 100644
--- a/package.json
+++ b/package.json
@@ -10,6 +10,7 @@
   "exports": {
     ".": "./_lib/index.js",
     "./components": "./_lib/components.js",
+    "./app/components/Link": "./_lib/app/components/Link.js",
     "./head": "./_lib/head.js",
     "./mdx-react": "./_lib/mdx-react.js"
   },
